src/Microsoft.DotNet.Build.Tasks.Packaging/src/PackageFiles/Packaging.targets

```diff
@@ -497,10 +497,10 @@
       <PackageSources Include="@(File)"
                       KeepMetadata="TargetPath;IsSourceCodeFile"
                       KeepDuplicates="false"
-                      Condition="'%(File.IsSourceCodeFile)'=='true'" />
+                      Condition="'%(File.IsSourceCodeFile)'=='true' AND '$(ExcludeSourceFiles)' != 'true'" />
       <!-- Add a placeholder source file if there are symbols but no sources. -->
       <PackageSymbolFiles Include="@(File)" Condition="'%(File.IsSymbolFile)'=='true'" />
-      <PackageSources Include="$(PlaceholderFile)" Condition="'@(PackageSymbolFiles)'!='' AND '@(PackageSources)'==''">
+      <PackageSources Include="$(PlaceholderFile)" Condition="'@(PackageSymbolFiles)'!='' AND '@(PackageSources)'=='' AND '$(ExcludeSourceFiles)' != 'true'">
```

Comment:
Would it be reasonable to put the condition on the `<PackageFile Include="@(PackageSources)" />` below? Would be slightly different if the repo supplied its own `PackageSources` items and I think preferable. (Although I don't know if repos do provide `PackageSources` items in practice.)

Reply:
That seems reasonable. I'll test out that change along w/ Wes' suggestions